apps/antidote/src/logging_vnode.erl

```diff
@@ -557,43 +559,47 @@ handle_command(
-                                true
+
+                        NewBucketOpId =
+                            case LogOperation#log_operation.op_type of
+                                OpType when (OpType == update) orelse (OpType == update_start) ->
+                                    Bucket =
+                                        (LogOperation#log_operation.log_payload)#update_log_payload.bucket,
+                                    BOpId = get_op_id(OpIdTable, {LogId, Bucket, MyDCID}),
+                                    #op_number{global = BGlobal} = BOpId,
+                                    NewBOpId = BOpId#op_number{global = BGlobal + 1},
+                                    true = update_ets_op_id(
+                                             {LogId, Bucket, MyDCID}, NewBOpId, OpIdTable
+                                            ),
+                                    NewOpId;
+                                OpType when OpType == commit ->
+                                    NewOpId;
+                                _ ->
+                                    NewOpId
                         end,
-                        ExternalOpNum = LogRecord#log_record.op_number,
-                        case insert_log_record(Log, LogId, LogRecord, EnableLog, Sync) of
+
+                        %% Previously we would not update the record, which caused wierd sequences for global
+                        %% Change opids before writing so that local equal remote local, and global is based
+                        %% on last global opid for local DC.
+                        OldOpId = LogRecord#log_record.op_number,
+                        OldBOpId= LogRecord#log_record.bucket_op_number,
+                        LogRecord2 = LogRecord#log_record{
+                                       op_number = OldOpId#op_number{global = NewOpId#op_number.global},
+                                       bucket_op_number = OldBOpId#op_number{global = NewBucketOpId#op_number.global}
+                                      },
+
+                        ExternalOpNum = LogRecord2#log_record.op_number,
+                        case insert_log_record(Log, LogId, LogRecord2, EnableLog, Sync) of
                             {ok, ExternalOpNum} ->
                                 %% Would need to uncomment this is local ops are sent to this function
                                 %% case IsLocal of
                                 %%     true -> inter_dc_log_sender_vnode:send(Partition, Operation);
                                 %%     false -> ok
                                 %% end,
-                                {AccErr, AccSucc ++ [NewOpId],
+                                {AccErr, [NewOpId | AccSucc],
                                     ordsets:add_element(Log, UpdatedLogs)};
                             {error, Reason} ->
                                 {AccErr ++ [{reply, {error, Reason}, State}], AccSucc, UpdatedLogs}
```

Comment:
Probably want to here also prepend for symmetry?

Reply:
Afaik resulting OpId was not used at all but the calling code.. so I changed the code, so that we return the largest OpId, instead of the smallest.